src/plexil-adapter/OwInterface.cpp

```diff
@@ -237,88 +237,44 @@ static void handle_joint_fault (Joint joint, int joint_index,
   handle_overtorque (joint, msg->effort[joint_index]);
 }
 
-void OwInterface::systemFaultMessageCallback
-(const  ow_faults::SystemFaults::ConstPtr& msg)
+template <typename T1, typename T2>
+void OwInterface::faultCallback (T1 msg_val, T2& fmap,
+                                 const string& component)
 {
-  // Publish all joint information for visibility to PLEXIL and handle any
-  // system-level fault messages.
-  uint64_t msg_val = msg->value;
-
-  for (auto const& entry : systemErrors){
+  for (auto const& entry : fmap) {
     string key = entry.first;
-    uint64_t value = entry.second.first;
-    bool b = entry.second.second;
-
-    if (checkFaultMessages("SYSTEM", msg_val, key, value, b)) {
-      systemErrors[key].second = !systemErrors[key].second;
+    T1 value = entry.second.first;
+    bool exists = entry.second.second;
+    bool faulty = msg_val & value == value;
```

Comment:
I think the intended priority is wrong .. you need to add brackets
```suggestion
    bool faulty = (msg_val & value) == value;
```

Reply:
Indeed, '==' has higher precedence than '&'.  I added the brackets.  I think it worked the old way only incidentally, because the different faults are all treated the same for now.